firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteSerializer.java

```diff
@@ -633,14 +633,30 @@ public com.google.firebase.firestore.core.Target decodeQueryTarget(QueryTarget t
   }
 
   StructuredAggregationQuery encodeStructuredAggregationQuery(
-      QueryTarget encodedQueryTarget, List<AggregateField> aggregateFields) {
+      QueryTarget encodedQueryTarget,
+      List<AggregateField> aggregateFields,
+      HashMap<String, String> aliasMap) {
     StructuredAggregationQuery.Builder structuredAggregationQuery =
         StructuredAggregationQuery.newBuilder();
     structuredAggregationQuery.setStructuredQuery(encodedQueryTarget.getStructuredQuery());
 
-    // We use a Set here to automatically remove duplicates.
-    Set<StructuredAggregationQuery.Aggregation> aggregations = new HashSet<>();
+    List<StructuredAggregationQuery.Aggregation> aggregations = new ArrayList<>();
+
+    HashSet<String> uniqueFields = new HashSet<>();
+    int aliasID = 1;
     for (AggregateField aggregateField : aggregateFields) {
+      // The code block below is used to deduplicate the same aggregate fields.
+      // If two aggregateFields are identical, their aliases would be the same.
+      // Therefore, when adding duplicated alias into uniqueFields, the size of uniqueFields
+      // won't increase, and we can skip this aggregateField processing.
+      final int count = uniqueFields.size();
+      uniqueFields.add(aggregateField.getAlias());
```

Comment:
Looks like `HashSet.add()` returns a boolean (true if the set did not already contain the specified element).